### PR TITLE
Add more 3.5 release notes

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Adds some new features that were not previously documented, a few more images, and some development notes based on issues from the rc.

## PR Checklist

- [n/a] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [x] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: doc/users/prev_whats_new/whats_new_3.5.0.rst [hunk 1]
```diff
@@ -14,6 +14,26 @@ the :ref:`github-stats`.
...
+-----------------------------------------------
+
+`.Figure.subplot_mosaic`, `.pyplot.subplot_mosaic` support *simple* Axes
+sharing (i.e., only `True`/`False` may be passed to *sharex*/*sharey*). When
+`True`, tick label visibility and Axis units will be shared.
+
+.. plot::
+    :include-source:
+
+    layout = [
```
Comment (timhoffm) [inline doc/users/prev_whats_new/whats_new_3.5.0.rst:27]: Change to `mosaic` as we've renamed the parameter to distinguish from constrained/tight layout?

File: doc/users/prev_whats_new/whats_new_3.5.0.rst [hunk 2]
```diff
@@ -130,9 +172,20 @@ A new keyword argument *interpolation_stage* is provided for
...
+   Example of the interpolation stage options.
+
 For more details see the discussion of the new keyword argument in
 :doc:`/gallery/images_contours_and_fields/image_antialiasing`.
 
+``imshow`` supports half-float arrays
+-------------------------------------
+
+The `~.axes.Axes.imshow` method now supports half-float arrays, i.e., NumPy
+arrays with dtype of ``np.float16``.
```
Comment (timhoffm) [inline doc/users/prev_whats_new/whats_new_3.5.0.rst:187]:
```suggestion
arrays with dtype ``np.float16``.
```

File: doc/users/prev_whats_new/whats_new_3.5.0.rst [hunk 3]
```diff
@@ -269,15 +351,115 @@ The `~mpl_toolkits.mplot3d.axes3d.Axes3D` class now has *computed_zorder*
...
+
 Allow changing the vertical axis in 3d plots
 ----------------------------------------------
 
 `~mpl_toolkits.mplot3d.axes3d.Axes3D.view_init` now has the parameter
 *vertical_axis* which allows switching which axis is aligned vertically.
 
+.. plot::
+
+    x = np.array([0, 1, 2, 4])
```
Comment (timhoffm) [inline doc/users/prev_whats_new/whats_new_3.5.0.rst:394]: A plot with more directional structure is more instructive, e.g.
~~~
    phi = np.linspace(0, np.pi, 18)
    x, y, z = np.hstack([
        np.array([r*np.sin(phi), r*np.cos(phi), np.full_like(phi, 8-r)]) for r in range(8)])
    
    
    fig, axs = plt.subplots(1, 3, figsize=(9, 3),
                            subplot_kw=dict(projection='3d'),
                            gridspec_kw=dict(wspace=0.5))
    for vert_a, ax in zip(['z', 'y', 'x'], axs):
        pc = ax.scatter(x, y, z, c=z)
        ax.view_init(azim=30, elev=30, vertical_axis=vert_a)
        ax.set(xlabel='x', ylabel='y', zlabel='z',
               title=f'vertical_axis={vert_a!r}')
~~~
grafik

**Review by timhoffm — APPROVED** [review]
Modulo one plot suggestion.

File: doc/users/prev_whats_new/whats_new_3.5.0.rst (hunk 3 above)
Comment (QuLogic) [reply doc/users/prev_whats_new/whats_new_3.5.0.rst]: Done, but with some tweaks for the size; 9 inches is too wide for the docs page.

File: doc/users/prev_whats_new/whats_new_3.5.0.rst [hunk 4]
```diff
@@ -179,6 +253,14 @@ argument, for example:
     ax.legend(title='Points',
               title_fontproperties={'family': 'serif', 'size': 20})
 
+``Text`` and ``TextBox`` added *parse_math* option
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (greglucas) [inline doc/users/prev_whats_new/whats_new_3.5.0.rst:257]:
```suggestion
--------------------------------------------------
```
Change to dashes here so the indentation is at the same level as all of the other items. Also above in the "A callback registry..." item.

**Review by greglucas — APPROVED** [review]
Looks good, just one suggestion about the heading levels.

File: doc/users/prev_whats_new/whats_new_3.5.0.rst (hunk 4 above)
Comment (QuLogic) [reply doc/users/prev_whats_new/whats_new_3.5.0.rst]: Done.

Comment (greglucas) [reply doc/users/prev_whats_new/whats_new_3.5.0.rst]: 👍  Feel free to self-merge after CI if you've got all the what's new entries you want in this one.

**Pull request merged**